addon/-private/system/model/internal-model.js

```diff
@@ -1218,6 +1219,10 @@ export default class InternalModel {
     @private
   */
   _changedKeys(updates) {
+    if (this.hasRecord && typeof this._record._changedKeys === 'function') {
```

Comment:
If we go this route, I think we should consider not branching here. Rather invoking some symbol on `this._record` which is then implemented by DS.Model as well. Thoughts?

Reply:
I agree your suggestion is a better approach

Reply:
@stefanpenner we can drop the function check, but we'll still need to branch depending on `this.hasRecord`